Cordova create fails if installed with "@" in the path

### Environment

Tested on Ubuntu (18.04.1), but the same issue is occurring on Mac.
Cordova version `8.1.2 (cordova-lib@8.1.1)`

### Steps to reproduce

```bash
mkdir build@2
cd build@2
npm install cordova
node_modules/.bin/cordova create hello com.example.hello HelloWorld
```

### What happens

```
Creating a new cordova project.
Unhandled "error" event. (  Error from Cordova Fetch: Error: npm: Command failed with exit code 1 Error output:
npm ERR! code ENOLOCAL
npm ERR! Could not install "../build@2/node_modules/cordova/node_modules/cordova-app-hello-world/index.js" as it is not a directory and is not a file with a name ending in .tgz, .tar.gz or .tar

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-12-05T15_21_50_141Z-debug.log)
```

### Why this happens

1. Cordova create wants to use the default template at `../build@2/node_modules/cordova/node_modules/cordova-app-hello-world/index.js`
1. It uses `isRemoteUri()` [[source](https://github.com/apache/cordova-create/blob/5e3aab6/index.js#L357-L359)] to determine whether the template location is a local path or remote URI. This notices the `@` and decides it's a remote URI.
1. Because the tempalte location is detected as remote, Cordova-create uses NPM to try and install it [[source](https://github.com/apache/cordova-create/blob/5e3aab6/index.js#L135-L139)].
1. NPM chokes on the path as it isn't a directory or tarball.

### Why this matters

We ran into this while automatically building Cordova projects using Jenkins.
* We're using a local installation of Cordova (not `npm add --global`) so we can easily pin different versions for each separate project and keep project configuration local to each Jenkins workspace.
* Jenkins uses `@` to split workspaces when running multiple concurrent builds. See `hudson.slaves.WorkspaceList` under https://wiki.jenkins.io/display/JENKINS/Features+controlled+by+system+properties.

We can work around the problem by reconfiguring Jenkins, but it feels like others will probably run into this problem. Is there a reason why paths with `@` are automatically assumed to be remote, even if they exist on the filesystem?

--- a/index.js
+++ b/index.js
@@ -22,7 +22,7 @@ const fs = require('fs-extra');
 var path = require('path');
 
 var tmp = require('tmp');
-var isUrl = require('is-url');
+const npa = require('npm-package-arg');
 var isObject = require('isobject');
 var pathIsInside = require('path-is-inside');
 var requireFresh = require('import-fresh');
@@ -168,5 +168,5 @@ function getSelfDestructingTempDir () {
 }
 
 function isRemoteUri (uri) {
-    return isUrl(uri) || uri.includes('@') || !fs.existsSync(uri);
+    return npa(uri).type !== 'directory';
 }

--- a/index.js
+++ b/index.js
@@ -88,7 +88,7 @@ function cordovaCreate (dest, opts = {}) {
             emit('log', 'Creating a new cordova project.');
 
             // Use cordova-fetch to obtain npm or git templates
-            if (isRemoteUri(opts.template)) {
+            if (needsToBeFetched(opts.template)) {
                 var target = opts.template;
                 emit('verbose', 'Using cordova-fetch for ' + target);
                 return fetch(target, getSelfDestructingTempDir(), {});
@@ -167,6 +167,6 @@ function getSelfDestructingTempDir () {
     }).name;
 }
 
-function isRemoteUri (uri) {
+function needsToBeFetched (uri) {
     return npa(uri).type !== 'directory';
 }

--- a/package.json
+++ b/package.json
@@ -29,8 +29,8 @@
     "cordova-fetch": "^2.0.0",
     "fs-extra": "^8.1.0",
     "import-fresh": "^3.1.0",
-    "is-url": "^1.2.4",
     "isobject": "^4.0.0",
+    "npm-package-arg": "^6.1.1",
     "path-is-inside": "^1.0.2",
     "tmp": "^0.1.0",
     "valid-identifier": "0.0.2"

--- a/spec/create.spec.js
+++ b/spec/create.spec.js
@@ -229,28 +229,22 @@ describe('when shit happens', function () {
     });
 });
 
-describe('cordova create isRemoteUri', () => {
-    let isRemoteUri;
+describe('cordova create needsToBeFetched', () => {
+    let needsToBeFetched;
 
     beforeEach(() => {
-        isRemoteUri = rewire('..').__get__('isRemoteUri');
+        needsToBeFetched = rewire('..').__get__('needsToBeFetched');
     });
 
     it('should recognize URLs as remote', () => {
-        expect(isRemoteUri('https://example.com/pkg/foo')).toBe(true);
+        expect(needsToBeFetched('https://example.com/pkg/foo')).toBe(true);
     });
 
     it('should recognize package@version as remote', () => {
-        const spec = 'foo@1';
-        spyOn(fs, 'existsSync').withArgs(spec).and.returnValue(false);
-
-        expect(isRemoteUri(spec)).toBe(true);
+        expect(needsToBeFetched('foo@1')).toBe(true);
     });
 
     it('should not detect paths as remote only because they include an @', () => {
-        const spec = '../foo@1';
-        spyOn(fs, 'existsSync').withArgs(spec).and.returnValue(true);
-
-        expect(isRemoteUri(spec)).toBe(false);
+        expect(needsToBeFetched('../foo@1')).toBe(false);
     });
 });

--- a/spec/create.spec.js
+++ b/spec/create.spec.js
@@ -18,6 +18,7 @@
 */
 
 const fs = require('fs-extra');
+const rewire = require('rewire');
 
 var path = require('path');
 
@@ -225,5 +226,31 @@ describe('when shit happens', function () {
             create(project, opts),
             new CordovaError('not a valid template')
         );
+    });
+});
+
+describe('cordova create isRemoteUri', () => {
+    let isRemoteUri;
+
+    beforeEach(() => {
+        isRemoteUri = rewire('..').__get__('isRemoteUri');
+    });
+
+    it('should recognize URLs as remote', () => {
+        expect(isRemoteUri('https://example.com/pkg/foo')).toBe(true);
+    });
+
+    it('should recognize package@version as remote', () => {
+        const spec = 'foo@1';
+        spyOn(fs, 'existsSync').withArgs(spec).and.returnValue(false);
+
+        expect(isRemoteUri(spec)).toBe(true);
+    });
+
+    it('should not detect paths as remote only because they include an @', () => {
+        const spec = '../foo@1';
+        spyOn(fs, 'existsSync').withArgs(spec).and.returnValue(true);
+
+        expect(isRemoteUri(spec)).toBe(false);
     });
 });

